types/dockerode/index.d.ts

```diff
@@ -16,9 +16,7 @@
 
 import * as stream from 'stream';
 import * as events from 'events';
-
-// The modem parameter is an instance of docker-modem, which is missing type declarations.
-// https://github.com/apocas/docker-modem
+import * as DockerModem from "docker-modem";
```

Comment:
I think this assumes the user has the `@types/docker-modem` package installed. Does it fail gracefully if not?

Reply:
I believe it adds `@types/docker-modem` as a dependency upon compilation, I've [generated](https://github.com/microsoft/DefinitelyTyped-tools/tree/master/packages/publisher#generate-packages-on-disk) it and it appears to be the case.

Reply:
That is correct, the DT publisher generates dependencies on other `@types/*` packages by inspecting source code.